archaius2-core/src/test/java/com/netflix/archaius/ProxyFactoryTest.java

```diff
@@ -231,4 +237,63 @@ public void testWithLongMap() {
         sub2 = withArgs.getChildren().get("2");
         Assert.assertEquals(789, sub2);
     }
+    
+    public static interface ConfigWithCollections {
+        List<Integer> getList();
+        
+        Set<Integer> getSet();
+        
+        SortedSet<Integer> getSortedSet();
+        
+        LinkedList<Integer> getLinkedList();
+    }
+    
+    @Test
+    public void testCollections() {
+        SettableConfig config = new DefaultSettableConfig();
+        config.setProperty("list", "5,4,3,2,1");
+        config.setProperty("set", "5,4,3,2,1");
+        config.setProperty("sortedSet", "5,4,3,2,1");
+        config.setProperty("linkedList", "5,4,3,2,1");
+        
+        PropertyFactory factory = DefaultPropertyFactory.from(config);
+        ConfigProxyFactory proxy = new ConfigProxyFactory(config, config.getDecoder(), factory);
+        ConfigWithCollections withCollections = proxy.newProxy(ConfigWithCollections.class);
+        
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getLinkedList()));
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getList()));
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getSet()));
```

Comment:
I'm actually creating a LinkedHashSet so the order is preserved (not sorted).  Not sure it's necessary but felt like it would be nice to maintain.   I can change this to a HashSet if you feel that's more appropriate.